scripts/deploy.sh

```diff
@@ -87,104 +47,69 @@ then
-    create "DripsHub" 'src/Managed.sol:ManagedProxy' "$DRIPS_HUB_LOGIC" "$DRIPS_HUB_ADMIN"
-    DRIPS_HUB=$DEPLOYED_ADDR
-fi
-
-if [ -z "$ADDRESS_DRIVER" ]; then
-    if [ -z "$ADDRESS_DRIVER_LOGIC" ]; then
-        NONCE=$(($(cast nonce $DEPLOYER) + 2))
-        ADDRESS_DRIVER=$(cast compute-address $DEPLOYER --nonce $NONCE | cut -d " " -f 3)
-        ADDRESS_DRIVER_ID=$(cast call "$DRIPS_HUB" 'nextDriverId()(uint32)')
-        send "Registering AddressDriver in DripsHub" \
-            "$DRIPS_HUB" 'registerDriver(address)(uint32)' "$ADDRESS_DRIVER"
-        create "AddressDriver logic" 'src/AddressDriver.sol:AddressDriver' \
-            "$DRIPS_HUB" "$CALLER" "$ADDRESS_DRIVER_ID"
-        ADDRESS_DRIVER_LOGIC=$DEPLOYED_ADDR
-    fi
-    create "AddressDriver" 'src/Managed.sol:ManagedProxy' \
-        "$ADDRESS_DRIVER_LOGIC" "$ADDRESS_DRIVER_ADMIN"
-    ADDRESS_DRIVER=$DEPLOYED_ADDR
-fi
-assertIsDriver "AddressDriver" "$ADDRESS_DRIVER"
-
-if [ -z "$NFT_DRIVER" ]; then
-    if [ -z "$NFT_DRIVER_LOGIC" ]; then
-        NONCE=$(($(cast nonce $DEPLOYER) + 2))
-        NFT_DRIVER=$(cast compute-address $DEPLOYER --nonce $NONCE | cut -d " " -f 3)
-        NFT_DRIVER_ID=$(cast call "$DRIPS_HUB" 'nextDriverId()(uint32)')
-        send "Registering NFTDriver in DripsHub" \
-            "$DRIPS_HUB" 'registerDriver(address)(uint32)' "$NFT_DRIVER"
-        create "NFTDriver logic" 'src/NFTDriver.sol:NFTDriver' \
-            "$DRIPS_HUB" "$CALLER" "$NFT_DRIVER_ID"
-        NFT_DRIVER_LOGIC=$DEPLOYED_ADDR
-    fi
-    create "NFTDriver" 'src/Managed.sol:ManagedProxy' "$NFT_DRIVER_LOGIC" "$NFT_DRIVER_ADMIN"
-    NFT_DRIVER=$DEPLOYED_ADDR
-fi
-assertIsDriver "NFTDriver" "$NFT_DRIVER"
-
-if [ -z "$SPLITS_DRIVER" ]; then
-    if [ -z "$SPLITS_DRIVER_LOGIC" ]; then
-        NONCE=$(($(cast nonce $DEPLOYER) + 2))
-        SPLITS_DRIVER=$(cast compute-address $DEPLOYER --nonce $NONCE | cut -d " " -f 3)
-        SPLITS_DRIVER_ID=$(cast call "$DRIPS_HUB" 'nextDriverId()(uint32)')
-        send "Registering ImmutableSplitsDriver in DripsHub" \
-            "$DRIPS_HUB" 'registerDriver(address)(uint32)' "$SPLITS_DRIVER"
-        create "ImmutableSplitsDriver logic" 'src/ImmutableSplitsDriver.sol:ImmutableSplitsDriver' \
-            "$DRIPS_HUB" "$SPLITS_DRIVER_ID"
-        SPLITS_DRIVER_LOGIC=$DEPLOYED_ADDR
-    fi
-    create "ImmutableSplitsDriver" 'src/Managed.sol:ManagedProxy' \
-        "$SPLITS_DRIVER_LOGIC" "$SPLITS_DRIVER_ADMIN"
-    SPLITS_DRIVER=$DEPLOYED_ADDR
+VERIFY=""
+if [ -n "$ETHERSCAN_API_KEY" ]; then
+    VERIFY="--verify"
 fi
-assertIsDriver "ImmutableSplitsDriver" "$SPLITS_DRIVER"
+DEPLOYER=$( \
+    forge create $VERIFY $WALLET_ARGS "src/Deployer.sol:Deployer" --constructor-args \
+        "$DRIPS_HUB_CYCLE_SECS" \
+        "$DRIPS_HUB_ADMIN" \
+        "$ADDRESS_DRIVER_ADMIN" \
+        "$NFT_DRIVER_ADMIN" \
+        "$SPLITS_DRIVER_ADMIN" \
+    | tee /dev/tty | grep '^Deployed to: ' | cut -d " " -f 3)
+deployment_detail() {
```

Comment:
I like your helper bash function. :sun_with_face: 
A parameter for defining the return type for the contract call.

We avoid the helper variable so it is good